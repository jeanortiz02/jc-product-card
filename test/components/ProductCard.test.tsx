
import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import { ProductCard } from '../../src/components';
import { product1 } from '../data/product';


describe('ProductCard', () => {
    
    test('debe mostrar el componente correctamente ', () => {
        
        const { container } = render(
            <ProductCard product={product1}>
                {
                    () => (
                        <h1>ProductCard</h1>
                    )
                }
            </ProductCard>
        )

       expect(container).toMatchSnapshot();
       
    });
    
    test('Debe incrementar el contador', async() => {
        const { container, getByText } = render(
            <ProductCard product={product1}>
                {
                    ({ count, increaseBy}) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{count}</span>
                            <button onClick={ () => increaseBy(1)}>+1</button>
                        </>
                    )
                }
            </ProductCard>
        )
    
    expect(container).toMatchSnapshot();
    const count = getByText('0');

    act(() => {
        fireEvent.click(getByText('+1'));
    });

    console.log(count.textContent);
    expect(count.textContent).toBe('1');
    });

});
  