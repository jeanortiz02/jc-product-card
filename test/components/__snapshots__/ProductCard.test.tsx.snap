// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Debe incrementar el contador 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
    <span>
      0
    </span>
    <button>
      +1
    </button>
  </div>
</div>
`;

exports[`ProductCard debe mostrar el componente correctamente  1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <h1>
      ProductCard
    </h1>
  </div>
</div>
`;
